pkg/kubernetes/deploy.js

```diff
@@ -77,6 +81,192 @@ define([
     }
 
     /*
+     * Runs the installed Nulecule app Image
+     */
+    function run_nulecule() {
+        var deferred = $.Deferred();
+        $("#deploy-app-dialog").dialog("failure", null);
+        $("#deploy-app-dialog").dialog("wait", deferred.promise());
+
+        validate_app_fields()
+            .fail(function(ex) {
+                deferred.reject(ex);
+                $("#deploy-app-dialog").dialog("failure", ex);
+            })
+            .done(function(fields) {
+                nulecule_client.run(fields.nulecule_image)
+                    .progress(function(msg) {
+                        deferred.notify(msg);
+                    })
+                    .done(function() {
+                        deferred.resolve();
+                        $('#deploy-app-dialog').modal('hide');
+                    })
+                    .fail(function(ex) {
+                        deferred.reject(ex);
+                        $("#deploy-app-dialog").dialog("failure", ex);
+                    });
+            });
+    }
+
+    /*
+     * Installs Nulecule app Image
+     */
+    function install_nulecule() {
+        var deferred = $.Deferred();
+        $("#deploy-app-dialog").dialog("failure", null);
+        $("#deploy-app-dialog").dialog("wait", deferred.promise());
+
+        validate()
+            .fail(function(exs) {
+                deferred.reject(exs);
+                $("#deploy-app-dialog").dialog("failure", exs);
+            })
+            .done(function(fields) {
+                nulecule_client.install(fields.nulecule_image)
+                    .progress(function(msg) {
+                        deferred.notify(msg);
+                    })
+                    .done(function(){
+                        var answerfile = nulecule_client.answers;
```

Comment:
WDYT, of returning this as a result and then passing it back in the run method. Or is that too complicated?